libraries/account-resolution/README.md

```diff
@@ -0,0 +1,130 @@
+};
+
+struct MyInstruction;
+impl TlvDiscriminator for MyInstruction {
+    // For ease of use, give it the same discriminator as its instruction definition
+    const TLV_DISCRIMINATOR: Discriminator = Discriminator::new([1; Discriminator::LENGTH]);
+}
+
+// Actually put it in the additional required account keys and signer / writable
+let extra_metas = [
+    AccountMeta::new(Pubkey::new_unique(), false),
+    AccountMeta::new(Pubkey::new_unique(), true),
+    AccountMeta::new_readonly(Pubkey::new_unique(), true),
+    AccountMeta::new_readonly(Pubkey::new_unique(), false),
+];
+
+// Assume that this buffer is actually account data, already allocated to `account_size`
+let account_size = ExtraAccountMetas::size_of(extra_metas.len()).unwrap();
+let mut buffer = vec![0; account_size];
+
+// Initialize the structure for your instruction
+ExtraAccountMetas::init_with_account_metas::<MyInstruction>(&mut buffer, &extra_metas).unwrap();
+
+// Off-chain, you can add the additional accounts directly from the account data
+let program_id = Pubkey::new_unique();
+let mut instruction = Instruction::new_with_bytes(program_id, &[], vec![]);
+ExtraAccountMetas::add_to_instruction::<MyInstruction>(&mut instruction, &buffer).unwrap();
+
+// On-chain, you can add the additional accounts *and* account infos
+let mut cpi_instruction = Instruction::new_with_bytes(program_id, &[], vec![]);
+
+// Include all of the well-known required account infos here first
+let mut cpi_account_infos = vec![]; 
+
+// Provide all "remaining_account_infos" that are *not* part of any other known interface
+let remaining_account_infos = &[]; 
+ExtraAccountMetas::add_to_cpi_instruction::<MyInstruction>(
+    &mut cpi_instruction,
+    &mut cpi_account_infos,
+    &buffer,
+    &remaining_account_infos,
+).unwrap();
+```
+
+For ease of use on-chain, `ExtraAccountMetas::init_with_account_infos` is also
+provided to initialize directly from a set of given accounts.
+
+## Motivation
+
+The Solana account model presents unique challeneges for program interfaces.
+Since it's impossible to load additional accounts on-chain, if a program requires
+additional accounts to properly implement an instruction, there's no clear way
+for clients to fetch these accounts.
+
+There are two main ways to fetch additional accounts, on-chain and off-chain.
+
+### Inside Program Logic
+
+To expose the additional accounts required, instruction interfaces can include
+supplemental instructions that just give the accounts required.
+
+For example, a `Transferable` interface, along with a `transfer` instruction,
+also requires implementations to expose a `get_additional_accounts_for_transfer`
+instruction.
+
+In the program implementation, this instruction writes the additional accounts
+into return data, making it easy for on-chain and off-chain clients to consume.
+
+See this
+[relevant sRFC](https://forum.solana.com/t/srfc-00010-additional-accounts-request-transfer-spec/122)
+for more information.
```

Comment:
I think this should be retitled `Comparison with Other methods` or something, since `get_additional_accounts_for_transfer` is not directly related. Unless I'm just missing something

Reply:
Yeah sure that's fine. I wanted to point out that there's really two main ways of resolving accounts that we all came up with -- either off-chain by reading account data, or on-chain via simulation. `get_additional_accounts_for_transfer` is meant to be part of the example of how you would do it using simulation. Either way, I'll move it to a separate section.